secrets/azurekeyvault/akv.go

```diff
@@ -219,20 +215,20 @@ var (
 // The "/{key-version}" suffix is optional; it defaults to the latest version.
 // See https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates
 // for more details.
-func OpenKeeper(client *keyvault.BaseClient, keyID string, opts *KeeperOptions) (*secrets.Keeper, error) {
+func OpenKeeper(client azcore.TokenCredential, keyID string, opts *KeeperOptions) (*secrets.Keeper, error) {
```

Comment:
Can we keep this as the Client? The URLOpener can create a default client.

Reply:
What's the purpose of passing in a default client here? We'd be discarding it to create one with the vault URL.

Reply:
The reason for this is similar to the rationale behind dependency injection. There are a million and three options for how to create a `Client`, so we want callers to be able to provide one explicitly instead of us creating one internally and thus only supporting options that we explicitly enable.

You are right though, that the constructor `NewClient` in the new package requires a vault URL, which we don't know until `OpenKeeper`.

One approach is to modify the constructors to take all of the parameters needed by `NewClient` (the credentials, plus `ClientOptions`). Another approach is to modify the constructors to take a function, e.g., `makeClient func(vaultURL string) *azkeys.Client`. It could default to `Dial`.

At this point I think it probably makes more sense for me to patch your Pull Request and modify it; I suspect that will be needed anyway to update the test goldens etc. I'll cc you on the PR to get your feedback.

Reply:
@AaronFriel  see PR #3175

Reply:
Thank you!